Better error message: panic: "->" is not a valid Git SHA

I get following output with `Git Town 10.0.2 (2023-11-07)`

```
c:\git-repositories\JabRef (main) > git-town sync
panic: "->" is not a valid Git SHA

goroutine 1 [running]:
github.com/git-town/git-town/v10/src/domain.NewSHA({0xc000232282?, 0xc000232245?})
        /home/kevlar/git-town/src/domain/sha.go:19 +0x88
github.com/git-town/git-town/v10/src/git.ParseVerboseBranchesOutput({0xc000232000, 0x2ed7})
        /home/kevlar/git-town/src/git/backend_commands.go:206 +0x4dd
github.com/git-town/git-town/v10/src/git.(*BackendCommands).BranchesSnapshot(0xc0001281f0)
        /home/kevlar/git-town/src/git/backend_commands.go:96 +0x90
github.com/git-town/git-town/v10/src/execute.LoadBranches({0xc0001281e0, 0x1, 0x0, 0x1, 0xc0002afe90, 0x1, 0x1, 0x0})
        /home/kevlar/git-town/src/execute/load_branches.go:18 +0xf9
github.com/git-town/git-town/v10/src/cmd.determineSyncConfig(0x0, 0xc0001281e0, 0x0)
        /home/kevlar/git-town/src/cmd/sync.go:136 +0x191
github.com/git-town/git-town/v10/src/cmd.executeSync(0x0?, 0x0?, 0x0)
        /home/kevlar/git-town/src/cmd/sync.go:70 +0xae
github.com/git-town/git-town/v10/src/cmd.syncCmd.func1(0xc0001ba500?, {0xeac8a0?, 0x4?, 0x941373?})
        /home/kevlar/git-town/src/cmd/sync.go:49 +0x68
github.com/spf13/cobra.(*Command).execute(0xc0001c2600, {0xeac8a0, 0x0, 0x0})
        /home/kevlar/git-town/vendor/github.com/spf13/cobra/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004300)
        /home/kevlar/git-town/vendor/github.com/spf13/cobra/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
        /home/kevlar/git-town/vendor/github.com/spf13/cobra/command.go:992
github.com/git-town/git-town/v10/src/cmd.Execute()
        /home/kevlar/git-town/src/cmd/core.go:33 +0x505
main.main()
        /home/kevlar/git-town/main.go:22 +0xc5
```

---

```cmd
c:\git-repositories\JabRef (main) > git branch -a | grep "\->"
  remotes/upstream/HEAD -> upstream/main
```

---

With `Git Town 9.0.1 (2023-07-29)` I have following output:

```
koppor@DESKTOP-KAK953S MINGW64 /c/git-repositories/jabref (main)
$ git-town sync

[main] git fetch --prune --tags

[main] git checkout remotes/upstream/main
Note: switching to 'remotes/upstream/main'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at c75506f6f4 Added parent field to Hayagriva YAML export (#10633)

[remotes/upstream/main] git fetch upstream remotes/upstream/main
fatal: couldn't find remote ref remotes/upstream/main

[remotes/upstream/main] git rebase upstream/remotes/upstream/main
fatal: invalid upstream 'upstream/remotes/upstream/main'

[remotes/upstream/main] git push -u origin remotes/upstream/main
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
To github.com:koppor/jabref.git
   2afd1f622a..c75506f6f4  upstream/main -> upstream/main

[remotes/upstream/main] git checkout main
Switched to branch 'main'
Your branch is up to date with 'origin/main'.

[main] git merge --no-edit origin/main
Already up to date.

[main] git merge --no-edit remotes/upstream/main
Already up to date.
```

I think, `git-town` should stop at the **first** fatal error, shouldn't it?

--- a/src/git/backend_commands.go
+++ b/src/git/backend_commands.go
@@ -450,7 +450,7 @@ func ParseVerboseBranchesOutput(output string) (gitdomain.BranchInfos, gitdomain
 			continue
 		}
 		parts := spaceRE.Split(line[2:], 3)
-		if parts[0] == "remotes/origin/HEAD" {
+		if strings.HasSuffix(parts[0], "/HEAD") {
 			continue
 		}
 		if len(parts) < 2 {

--- a/src/git/backend_commands_test.go
+++ b/src/git/backend_commands_test.go
@@ -638,6 +638,8 @@ func TestBackendCommands(t *testing.T) {
   remotes/origin/branch-5      55555555 Commit message 5
   remotes/origin/HEAD          -> origin/initial
   remotes/origin/main          024df944 Commit message on main (#1234)
+	remotes/upstream/HEAD        -> upstream/main
+
 `[1:]
 			want := gitdomain.BranchInfos{
 				gitdomain.BranchInfo{

--- a/src/git/backend_commands_test.go
+++ b/src/git/backend_commands_test.go
@@ -638,7 +638,7 @@ func TestBackendCommands(t *testing.T) {
   remotes/origin/branch-5      55555555 Commit message 5
   remotes/origin/HEAD          -> origin/initial
   remotes/origin/main          024df944 Commit message on main (#1234)
-	remotes/upstream/HEAD        -> upstream/main
+  remotes/upstream/HEAD        -> upstream/main
 
 `[1:]
 			want := gitdomain.BranchInfos{

